panel_segmentation/utils.py

```diff
@@ -0,0 +1,709 @@
+    if not isinstance(longitude, float):
+        raise TypeError("longitude variable must be of type float.")
+    if not isinstance(google_maps_api_key, str):
+        raise TypeError("google_maps_api_key variable must be "
+                        "of type string.")
+    # return response object
+    r = requests.get(
+        "https://maps.googleapis.com/maps/api/geocode/json?latlng=" +
+        str(latitude) + "," + str(longitude) + "&key=" + google_maps_api_key,
+        verify=False)
+    # Raise an exception if address is not successfully returned
+    if r.status_code != 200:
+        raise ValueError("Response status code " +
+                         str(r.status_code) +
+                         ": Address not pulled successfully from API.")
+    data = r.json()
+    address = data["results"][0]["formatted_address"]
+    return address
+
+
+def generateSatelliteImageryGrid(northwest_latitude, northwest_longitude,
+                                 southeast_latitude, southeast_longitude,
+                                 google_maps_api_key,
+                                 file_save_folder,
+                                 zoom_level=18,
+                                 lat_lon_distance=0.00145,
+                                 number_allowed_images_taken=6000):
+    """
+    Take satellite images via the Google Maps API in a grid fashion for a large
+    area, and save the associated images to a folder. The associated images
+    can then be used to feed into different models to assess a larger area.
+
+    Please note rates for running the Google Maps API here:
+        https://developers.google.com/maps/billing-and-pricing/pricing
+
+    Parameters
+    ----------
+    northwest_latitude : float
+        Latitude coordinate on the northwest corner of the area we wish
+        to scan.
+    northwest_longitude : float
+        Longitude coordinate on the northwest corner of the area we wish
+        to scan.
+    southeast_latitude : float
+        Latitude coordinate on the southeast corner of the area
+        we wish to scan.
+    southeast_longitude : float
+        Longitude coordinate on the southeast corner of the area
+        we wish to scan.
+    google_maps_api_key : str
+        API key for Google Maps API
+    file_save_folder : str
+        Folder path for which to save all of the images
+    zoom_level: int, default 18
+        Zoom level of the image.
+    lat_lon_distance : float, default 0.00145
+        Distance to traverse between images, in terms of lat-long
+        degree distance. For example, default distance of 0.00145 degrees
+        equates to approx. 161 meters.
+    number_allowed_images_taken : int, default 6000
+        Number of allowed images for the Google Maps API to take before
+        stopping. If we pull too many images in one go, google may flag
+        this as webscraping so it's advised to not pull too many images
+        at once.
+
+    Returns
+    -------
+    None.
```

Comment:
We missed the variable returned here (prob my fault honestly): grid_location_list